stl/inc/tuple

```diff
@@ -223,6 +269,26 @@ public:
         return false;
     }
 #endif // ^^^ !defined(__cpp_lib_concepts) ^^^
+
+#if _HAS_CXX23 && defined(__cpp_lib_concepts)
+    template <_Tuple_like _Other>
+#ifndef __clang__ // TRANSITION, MSVC cannot figure out which operator== it should pick
+        requires _Different_from<_Other, tuple>
+#endif // __clang__
+    _NODISCARD_FRIEND constexpr bool operator==(const tuple&, const _Other&) noexcept /* strengthened */ {

```

Comment:
I assume this is `reject-on-valid` type of bug. Reduced case:

```c++
template <class...> class MyTuple;

template <> class MyTuple<> {
public:
  template <class Other> friend bool operator==(const MyTuple &, const Other &);
};

template <class This, class... Rest>
class MyTuple<This, Rest...> : private MyTuple<Rest...> {
public:
  template <class Other> MyTuple(Other &&right);

  template <class Other> friend bool operator==(const MyTuple &, const Other &);
};

MyTuple<int, int> decl();
using F = decltype(0 == decl());
```

Compiler explorer: https://godbolt.org/z/f4c8hzeeE

(please feel free to report it)

Reply:
I've reported DevCom-10265322. I think wokaround could be adding another `operator==` with reversed parameter order.